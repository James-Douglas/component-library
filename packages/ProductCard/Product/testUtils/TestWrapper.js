import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
