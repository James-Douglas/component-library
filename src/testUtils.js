// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
