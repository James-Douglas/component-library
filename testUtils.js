import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
