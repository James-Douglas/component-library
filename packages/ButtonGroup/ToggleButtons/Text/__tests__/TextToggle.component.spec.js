import React from 'react';
import { useTracking } from 'react-tracking';
import { fireEvent, render } from '../../../../../testUtils';
import TextToggle from '../TextToggle.component';

describe('TextToggle', () => {
  it('renders with props', () => {
    const { container } = render(
      <TextToggle id="test" trackingLabel="test" value="test" title="hello" />,
    );
    expect(container.innerHTML).toMatchSnapshot();
  });
  describe('interaction tracking', () => {
    it('tracks clicks', () => {
      const { trackEvent } = useTracking();
      const { container } = render(
        <TextToggle id="test" trackingLabel="test" value="test" title="hello" />,
      );
      const input = container.querySelector('input');
      fireEvent.click(input);
      expect(trackEvent.mock.calls[0][0]).toStrictEqual({
        interaction: {
          ixn_action: 'Click',
          ixn_label: '',
          ixn_object: 'Toggle Buttons',
          ixn_type: 'TextToggle',
          ixn_value: 'test',
        },
      });
    });
  });
});
