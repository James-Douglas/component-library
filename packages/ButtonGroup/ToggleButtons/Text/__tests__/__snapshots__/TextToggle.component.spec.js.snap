// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextToggle renders with props 1`] = `"<div class=\\"sc-AxirZ klQxSA\\"><input tabindex=\\"0\\" id=\\"test\\" type=\\"radio\\" name=\\"\\" class=\\"sc-AxiKw botYzr\\" value=\\"test\\"><label for=\\"test\\" class=\\"sc-AxhCb DNJow\\"><div class=\\"sc-AxhUy eQbMTQ\\"><div class=\\"sc-AxgMl diSfKa\\"><div class=\\"sc-AxheI bwBlNt\\"><div class=\\"sc-Axmtr eYgBli\\"><p class=\\"sc-AxjAm ckNpsi\\">hello</p></div></div></div></div></label></div>"`;
